src/utils/general-helpers.js

```diff
@@ -26,18 +24,49 @@ async function executeBeforeBuildPlugins(programInfo, { createContent }) {
  */
 async function executeAfterBuildPlugins(programInfo) {
   /** After Build plugins */
+  console.log('\n>> Executing afterBuild plugins');
   for (const pluginPath of programInfo.abellConfig.plugins) {
     const currentPlugin = require(pluginPath);
     if (currentPlugin.afterBuild) {
+      if (programInfo.logs === 'complete') {
+        console.log('> Plugin ' + path.relative(process.cwd(), pluginPath));
+      }
+      await currentPlugin.afterBuild(programInfo);
+    }
+  }
+}
+
+/**
+ * executes beforeHTMLWrite plugins
+ * @param {string} htmlOutput HTML code in string
+ * @param {ProgramInfo} programInfo
+ */
+async function executeBeforeHTMLWritePlugins(htmlOutput, programInfo) {
+  for (const pluginPath of programInfo.abellConfig.plugins) {
+    const currentPlugin = require(pluginPath);
```

Comment:
We are requiring plugins on every build step we do. For example we also require them here:
https://github.com/abelljs/abell/blob/f498c38c7421f5d15f5809b4160b7403b9169bcb/src/utils/general-helpers.js#L12

That is not so good for perf imo as importing takes resources. We can refactor this to reuse the imported instances later.

Reply:
Correct me if I'm wrong but I think `require` manages it's own cache by default. So if I call require on same url twice, first time it will read and load the object and will store it in `require.cache` and next time it will read it straight from cache. So these plugins are only being read once.

Reply:
Ohhh, I forgot about it caches things 😅. Yeah, this is fine to do